src/services/async-component-resolver.ts

```diff
@@ -0,0 +1,23 @@
+import { Type, isFunction } from '@angular/core/src/facade/lang';
+import { Injectable, ComponentFactory, ComponentResolver } from '@angular/core';
+
+/**
+ * Component resolver that can load components lazily
+ *
+ * TODO: Update to use AppModule lazy loading
+ */
+@Injectable()
+export class AsyncComponentResolver implements ComponentResolver {
+  constructor(private _resolver: ComponentResolver) {}
+
+  resolveComponent(componentType: string|Type): Promise<ComponentFactory<any>> {
+    if (isFunction(componentType) && (<any>componentType).name === 'component') {
```

Comment:
I'm not sure I fully understand how this is able to distinguish between functions that return promises and component classes. Is this minification safe?